### DISCUSSION TITLE
Working with multiple RDL2 scene files and external data

### DISCUSSION THREAD

**Discussion by ramenguy99** [dreamworksanimation/openmoonray#28] [Q&A] [answered]
Hi everyone, congrats for the open source launch!

I'm looking through the documentation of the RDL2 scene format and one thing that isn't clear to me is what is the intended way to deal with scenes that include external data such as large meshes.

Is it possible to break down a scene in multiple RDL2 files and combine them in some way? Does this work directly with Lua imports and is this handled by the binary format? Is it possible to mix the two in some way, for example by having a small text scene including a bunch of binary RDL2 files to create a more complex scene?

**Comment by kjeffery** [comment, marked as answer]
Hi @ramenguy99 !

Several RDL2 geometry primitives reference external files, e.g., `UsdGeometry`.

That said, it is possible to give MoonRay multiple RDL files as input. For example, we frequently supply an RDLA file (for parameters like `SceneVariables`) and an RDLB file for things like geometry.

```
moonray -in scene.rdla -in scene.rdlb
```

**Reply by ramenguy99** [reply]
Thanks that's exactly what I was looking for!

**Comment by kjeffery** [comment]
I wanted to clarify because I don't think it's obvious:

* rdla is a text-based RDL file
* rdlb is a binary RDL file

Since they happen to be labeled "a" (for ASCII) and "b" (for binary), it seems like a versioning system.

**Comment by kjeffery** [comment]
https://docs.openmoonray.org/developer-reference/scene_rdl2-library/ has further examples and documentation about using multiple rdl files. It's also possible to convert between the text-based and binary formats through `rdl2_convert`.

**Comment by jlanz** [comment]
I'll also mention this page ( see Inputs section ):
https://docs.openmoonray.org/user-reference/tools/moonray/

It briefly discusses this, but it probably needs some example RDLA snippets to show how you can append/overwrite attributes on subsequent input files.
